src/RequestorPages/PastPosts.js

```diff
@@ -75,70 +76,135 @@ const Job = ({job: {title, jobType, startDateTime, endDateTime, location, reques
             let data = querySnapShot.val() ? querySnapShot.val() : {};
             let jobItems = {...data};
+            var jobArray = [];
+            Object.keys(jobItems).map((key) => {
+                if (jobItems[key].title != null) {
+                    jobArray.push(jobItems[key]);
+                }
+            })
             this.setState({
-            jobs: sortBy(jobItems, 'date'),
+                jobs: jobArray.sort(function compare(a, b) {
+                    var dateA = new Date(a.startDateTime);
+                    var dateB = new Date(b.startDateTime);
+                    return dateA - dateB;
+                }),
             });
         });
 
         db.ref('/users').orderByChild('username').on('value', querySnapShot => {
-          let data = querySnapShot.val() ? querySnapShot.val() : {};
-          let userItems = {...data};
-          this.setState({
-              allUsers: sortBy(userItems, 'username'),
-          });
-      }); 
+            let data = querySnapShot.val() ? querySnapShot.val() : {};
+            let userItems = {...data};
+            this.setState({
+                allUsers: sortBy(userItems, 'username'),
+            });
+        });
     }
 
     getActiveUser(userKeys) {
-      let value = this.context;
-      for (var i = 0; i < userKeys.length; i++) {
-        var curr = this.state.allUsers[userKeys[i]];
-        if (curr.username == value["username"]) {
-          activeUser.username = curr.username;
-          activeUser.trustedUsers = curr.trustedUsers;
+        let value = this.context;
+        for (var i = 0; i < userKeys.length; i++) {
+            var curr = this.state.allUsers[userKeys[i]];
+            if (curr.username == value["username"]) {
+                activeUser.username = curr.username;
+                activeUser.trustedUsers = curr.trustedUsers;
+            }
         }
-      }
     }     
 
     render () {
         let jobsKeys = Object.keys(this.state.jobs);
         this.getActiveUser(Object.keys(this.state.allUsers));
         return (
-          <SafeAreaView style={styles.safeContainer}>
-            <ScrollView style={styles.scrollView}>
-            <View style={styles.container}>
-              {jobsKeys.length > 0 ? (
-                jobsKeys.map(key => (
-                  <Job
-                    key={key}
-                    id={key}
-                    job={this.state.jobs[key]}
-                  />
-                ))
-              ) : (
-                    <Text>No previous jobs</Text>
-              )}
-            </View>
-            </ScrollView>
-          </SafeAreaView>
+            <SafeAreaView style={styles.safeContainer}>
+                <ScrollView style={styles.scrollView}>
+                    {this.state.jobs.length > 0 ? (
+                        this.state.jobs.map((currentJob) => {
+                            if (currentJob.title != null)
+                            {
```

Comment:
Why is all of this logic now just happening in the render method instead of passing the args to the Job object, like it was before?

Reply:
I mentioned this in my initial message about my changes, but I had to switch the jobs data stored in the state to be an array of the jobs so that it is sortable. After looking online for guides to passing in props and using constants like we have for Jobs, I still wasn't able to figure out the proper way to call the Job object to render the Job (there wasn't any error, it just wasn't showing, even when I tried to make another kind of constant like it and pass the args there)

Reply:
With it being an array of jobs, have you tested the accept -> delete -> reaccept job sequence within the same user session? As in login as a volunteer, go to the job board, accept a job, go to assigned jobs, "delete" the acceptance of the job, and then re-sign-up for it on the job board? This sequence was working before without issue and I want to make sure that it'll still work when this branch is merged in.

Reply:
I don't think that sequence is working on this branch, but with so many changes to job board, I'm not sure what is causing it. I just tried switching to the main branch and logging in as a volunteer, but no jobs were shown on any of the pages for any of the volunteers I logged in as (Paula, Elena, and Clara). Is that what should be expected?